compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java

```diff
@@ -1895,18 +1896,28 @@ void generateNegateIns(BIRNonTerminator.UnaryOP unaryOp) {
     }
 
     void generateNewTypedescIns(BIRNonTerminator.NewTypeDesc newTypeDesc) {
-
+        List<BIROperand> closureVars = newTypeDesc.closureVars;
         BType type = newTypeDesc.type;
+        if (type.tag == TypeTags.RECORD && closureVars.size() == 0 && type.tsymbol != null) {
```

Comment:
Can type.tsymbol be null?

Reply:
It seems to coming null for `map` this is good for me because I anyway needed to avoid that.
Maybe I am relaying on a bug